File: 2021/02/One.java
import java.io.BufferedReader;
import java.io.FileReader;

public class One {

    public static void main(String[] args) {
        int h = 0;
        int v = 0;

        try {
            BufferedReader reader = new BufferedReader(new FileReader("2021/02/input.txt"));
            String line = null;
            while ((line = reader.readLine()) != null) {
                String[] lineSplit = line.split(" ");
                int dist = Integer.parseInt(lineSplit[1]);
                switch (lineSplit[0]) {
                    case "forward":
                        h += dist;
                        break;
                
                    case "up":
                        v -= dist;
                        break;

                    case "down":
                        v += dist;
                        break;

                    default:
                        System.err.println("unknown command: " + lineSplit[0]);
                        break;
                }
            }
            reader.close();
        } catch (Exception e) {
            System.out.println(e.getStackTrace());
        }

        System.out.println(h * v);
    }
}

File: 2021/02/one.py
h = v = 0
for l in open('2021/02/input.txt'):
    cmd, dist = l.split()
    dist = int(dist)
    if cmd == 'forward':
        h += dist
    elif cmd == 'up':
        v -= dist
    elif cmd == 'down':
        v += dist
    else:
        raise Exception('unknown command: ' + cmd)
print(h, v, h*v)
